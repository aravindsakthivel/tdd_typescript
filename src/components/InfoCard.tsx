import { FC } from 'react';
import { Typography, Card } from 'antd';
import { OverviewInfoCardInterface } from './Interfaces/overviewInfoCardInterface';
import './infoCard.css';
const { Text, Title } = Typography;

const InfoCard: FC<OverviewInfoCardInterface> = ({
  title = 'User Onboarded this month',
  onBoardCount,
}) => {
  return (
    <Card style={{ width: 218, height: 108, background: '#492D66', border: 0 }} className="referral-info-card">
      <Text type="secondary" data-testid="info-card-title" style={{color: '#ffffff'}}>
        {title}
      </Text>
      <Title level={3} style={{ marginTop: '5px' }} data-testid="info-card-onboard-count">
        {onBoardCount}
      </Title>
    </Card>
  );
};

export { InfoCard };
