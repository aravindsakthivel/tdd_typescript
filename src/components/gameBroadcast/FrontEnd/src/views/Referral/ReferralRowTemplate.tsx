import React, { ReactElement } from 'react';
import { Row, Col, Typography } from 'antd';
import { ReferralRowTemplateInterface } from './ReferralInterfaces';

const { Title } = Typography;

function ReferralRowTemplate({
  children,
  title = 'None',
  last = false,
}: ReferralRowTemplateInterface): ReactElement {
  return (
    <Row
      className={`referral-row-template-class general-background-border ${
        last && 'no-border-last'
      }`}>
      <Col span={24}>
        <Title level={5} data-testid="referral-row-title">
          {title}
        </Title>
      </Col>
      <Col span={23} data-testid="referral-row-children">
        {children}
      </Col>
    </Row>
  );
}

export default ReferralRowTemplate;
