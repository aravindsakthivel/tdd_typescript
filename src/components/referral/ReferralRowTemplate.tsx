import { FC, ReactNode, ReactElement } from 'react';
import { Row, Col, Typography } from 'antd';
import '../customClassStyle.css';

const { Title } = Typography;

interface ReferralRowTemplateInterface {
  children?: ReactNode;
  title?: string;
  last?: boolean;
}

const ReferralRowTemplate: FC<ReferralRowTemplateInterface> = ({
  children,
  title = 'None',
  last = false,
}): ReactElement => {
  return (
    <Row
      className={`referral-row-template-class general-background-border ${
        last && 'no-border-last'
      }`}
    >
      <Col span={24}>
        <Title level={5} data-testid="referral-row-title">
          {title}
        </Title>
      </Col>
      <Col span={23} data-testid="referral-row-children">
        {children}
      </Col>
    </Row>
  );
};

export { ReferralRowTemplate };
