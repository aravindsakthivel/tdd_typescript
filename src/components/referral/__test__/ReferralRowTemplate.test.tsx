import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { mockConfig } from '../../mock';
import { ReferralRowTemplate } from '../ReferralRowTemplate';
import { InfoCard } from '../../InfoCard';

afterEach(cleanup);
beforeEach(() => {
  mockConfig();
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ReferralRowTemplate title={'none'} />, div);
});

it('the component consist of Title and Children Node', () => {
  let topTestTitle = 'test-text';
  const title = 'User Onboarded this month';
  const count = '320';
  const { getByTestId } = render(
    <ReferralRowTemplate
      title={topTestTitle}
      children={<InfoCard title={title} onBoardCount={count} />}
    />,
  );
  expect(getByTestId('referral-row-title')).toHaveTextContent(topTestTitle);
  expect(getByTestId('info-card-title')).toHaveTextContent(title);
  expect(getByTestId('info-card-onboard-count')).toHaveTextContent(count);
});

it('matches snapshot with title', () => {
  const tree = renderer.create(<ReferralRowTemplate title={'none'} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('matches snapshot with title and children', () => {
  let topTestTitle = 'test-text';
  const title = 'User Onboarded this month';
  const count = '320';
  const tree = renderer
    .create(
      <ReferralRowTemplate
        title={topTestTitle}
        children={<InfoCard title={title} onBoardCount={count} />}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
