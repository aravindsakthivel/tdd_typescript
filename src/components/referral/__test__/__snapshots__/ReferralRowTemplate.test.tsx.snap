// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with title 1`] = `
<div
  className="ant-row referral-row-template-class general-background-border false"
  style={Object {}}
>
  <div
    className="ant-col ant-col-24"
    style={Object {}}
  >
    <h5
      className="ant-typography"
      data-testid="referral-row-title"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      none
    </h5>
  </div>
  <div
    className="ant-col ant-col-23"
    data-testid="referral-row-children"
    style={Object {}}
  />
</div>
`;

exports[`matches snapshot with title and children 1`] = `
<div
  className="ant-row referral-row-template-class general-background-border false"
  style={Object {}}
>
  <div
    className="ant-col ant-col-24"
    style={Object {}}
  >
    <h5
      className="ant-typography"
      data-testid="referral-row-title"
      style={
        Object {
          "WebkitLineClamp": undefined,
        }
      }
    >
      test-text
    </h5>
  </div>
  <div
    className="ant-col ant-col-23"
    data-testid="referral-row-children"
    style={Object {}}
  >
    <div
      className="ant-card ant-card-bordered referral-info-card"
      style={
        Object {
          "background": "#492D66",
          "border": 0,
          "height": 108,
          "width": 218,
        }
      }
    >
      <div
        className="ant-card-body"
        style={Object {}}
      >
        <span
          className="ant-typography ant-typography-secondary"
          data-testid="info-card-title"
          style={
            Object {
              "WebkitLineClamp": undefined,
              "color": "#ffffff",
            }
          }
        >
          User Onboarded this month
        </span>
        <h3
          className="ant-typography"
          data-testid="info-card-onboard-count"
          style={
            Object {
              "WebkitLineClamp": undefined,
              "marginTop": "5px",
            }
          }
        >
          320
        </h3>
      </div>
    </div>
  </div>
</div>
`;
